### PR TITLE
Milestone 08:  Integration specs and fix n+1 problems

### PR DESCRIPTION
**Pull request by FaranoshAmini**
## In this PR we have  completed the following tasks:

- [x]  Made sure that the N+1 problem is solved when fetching all posts and their comments for a user.
- [x]  Used Capybara to write some integration tests for the:
- [x]  User Index page
- [x]  User Show page
- [x]  User Post Index page
- [x]  Post Show page
- Screenshots before N+1 and after
before solving n+1
after solving n+1
- Screenshot the test
test

### CODE REVIEW

File: spec/views/users_index_spec.rb [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+#     end
+#     it 'Display the number of posts each user has written' do
+#       expect(page).to have_content('Number of posts: 0')
+#     end
+#     it 'redirects to the user show page' do
+#       click_link('Farah', match: :prefer_exact)
+#       expect(page).to have_content('Bio:')
+#     end
+#   end
+# end
```
Comment (alexander16108) [inline spec/views/users_index_spec.rb]: please remove commented code from your code base as it is not a best practice to leave commented code in your code base 😄

File: spec/views/posts_index_spec.rb [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+    it 'should display the number of likes of the post' do
+      expect(page).to have_content(@posts.first.likes_counter)
+    end
+
+    it 'should include a section for paginations' do
+      expect(page).to have_content('Pagination')
+    end
+  end
+end
```
Comment (alexander16108) [inline spec/views/posts_index_spec.rb]: your should kindly include the test case below 😄 

`` Tests that redirects to the post's show page when you click on a post.``

**Review by alexander16108 — CHANGES_REQUESTED** [review]
Hi @FaranoshAmini @george827 @shegy28 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights

- Your project looks awesome👏 ✔️
- Your PR is professional and well-detailed ✔️
- I love the fact that you collaborated to a great extend as shown by the structure of the commits messages 👏 ✔️
- It's a good thing that you were able to get your linter checks to pass ✔️

## Required Changes ♻️
_Check the comments under the review._

- [x]  please remove the live demo link completely as it currently serves no purpose 👍🏾 as there is no link to publish

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by victorokoroji — APPROVED** [review]
Hi @FaranoshAmini , 

# ✔️   Status: APPROVED 🏆 🎉 🎊 

You have done a great job on this project, keep it up 👍 🥇. Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
congratulations!  🎉💯🌟 

## Top highlight 🎯

- The Pull Request is following the GitHub Flow ✅ 
- The PR title and description are professional and descriptive ✅ 
- Required changes implemented ✅ 
 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by FaranoshAmini** [comment]
@alexander16108, and @vickymarz  Thanks for your kind review and feedback.

**Pull request merged**